### PR TITLE
Problem: overlap option is not set to jobs

### LINKED ISSUE #71: Problem: setting overlap doesn't have any effect
Hi, thanks for making this library. I'm having trouble configuring the tasks to not overlap and I'm not sure if I'm doing something wrong with the configuration or if it's a bug.

This is the configuration I used:

``` elixir
config :quantum,
  default_overlap: false,
  cron: [
    snapshot_cleanup: [
      overlap: false,
      task: "EvercamMedia.Snapshot.Cleanup.init",
      schedule: "* * * * *",
    ]
  ]
```

Checking from `iex` everything seems to be configured correctly:

``` elixir
iex(1)> Application.get_env(:quantum, :cron)
[snapshot_cleanup: [overlap: false, task: "EvercamMedia.Snapshot.Cleanup.init",
  schedule: "* * * * *"]]

iex(2)> Application.get_env(:quantum, :default_overlap) 
false
```

But the task is still set to `overlap: true`:

``` elixir
iex(3)> Quantum.jobs                                    
[snapshot_cleanup: %Quantum.Job{args: [], name: :snapshot_cleanup,
  nodes: [:nonode@nohost], overlap: true, pid: #PID<0.1041.0>,
  schedule: "* * * * *", state: :active,
  task: {"EvercamMedia.Snapshot.Cleanup", "init"}}]
```

I'm running the latest version of the master branch. Any suggestions?

### PR DESCRIPTION
**Pull request by c-rack**
Solution: add test case and add overlap option in normalize step

Fixes #71

### CODE REVIEW

File: lib/quantum/normalizer.ex [hunk 1]
```diff
@@ -110,6 +110,7 @@ defmodule Quantum.Normalizer do
       schedule: extract(:schedule, opts),
       task: extract(:task, opts),
       args: extract(:args, opts, []),
+      overlap: extract(:overlap, opts, Application.get_env(:quantum, :default_overlap, true)),
```
Comment (theoretick) [inline lib/quantum/normalizer.ex]: I'm a bit confused on this fix. wont the defaults already be set within the `Quantum.Job` 's defstruct definition?  And if I'm missing something and they are not, should we have the other assorted opts like `args` defaulting to `Application.get_env` as well here?

Comment (c-rack) [inline lib/quantum/normalizer.ex]: You are right. It was just a quick fix to resolve this issue.
There is a more elegant way which I plan to submit on the weekend.

Comment (c-rack) [inline lib/quantum/normalizer.ex]: Or maybe tomorrow :-)

Comment (theoretick) [inline lib/quantum/normalizer.ex]: :+1:

**Pull request merged**